pyqtorch/circuit.py

```diff
@@ -215,6 +215,112 @@ def sample(probs: Tensor) -> Counter:
             return list(map(lambda p: sample(p), probs))
 
 
+class DropoutQuantumCircuit(QuantumCircuit):
+    def __init__(
+        self,
+        n_qubits: int,
+        operations: list[Module],
+        dropout_mode: DropoutMode = DropoutMode.ROTATIONAL,  # type: ignore
+        dropout_prob: float = 1.0,
+    ):
+        super().__init__(n_qubits, operations)
+        self.dropout_mode = dropout_mode
+        self.dropout_prob = dropout_prob
+
+        self.dropout_fn = getattr(self, dropout_mode)
+
+    def forward(
+        self, state: State, values: dict[str, Tensor] | ParameterDict = {}
+    ) -> State:
+        if self.training:
+            state = self.dropout_fn(state, values)
+        else:
+            for op in self.operations:
+                state = op(state, values)
+        return state
+
+    def rotational_dropout(
+        self, state: State = None, values: dict[str, Tensor] | ParameterDict = {}
+    ) -> State:
+        """Randomly drops entangling rotational gates.
+
+        Args:
+            state (State, optional): pure state vector . Defaults to None.
+            values (dict[str, Tensor] | ParameterDict, optional): gate parameters. Defaults to {}.
+
+        Returns:
+            State: pure state vector
+        """
+        for op in self.operations:
+            if hasattr(op, "param_name"):
+                if values[op.param_name].requires_grad:
+                    keep = 1 - bernoulli(tensor(self.dropout_prob))  # type: ignore
+                    if keep:
+                        state = op(state, values)
+
+                else:
+                    state = op(state, values)
+            else:
+                state = op(state, values)
+
+        return state
```

Comment:
``` # mild suggestion if it works
for op in self.operations:
    if hasattr(op, "param_name") and values[op.param_name].requires_grad and 1 - bernoulli(tensor(self.dropout_prob)):
        return state
    else:
        return op(state, values) # new state
```

Reply:
Potentially, if useful this can be applied to the code of the other types of dropout.